init-interfaces.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+set -ex
+
+is_con_exists() {
+	local con_name=$1
+	if [[ $(nmcli -t -f NAME con show | grep -w "$con_name") ]]; then
+		echo "true"
+	else
+		echo "false"
+	fi
+}
+
+is_con_active() {
+	local con_name=$1
+	if [[ $(nmcli -t -f NAME con show --active | grep -w "$con_name") ]]; then
+		echo "true"
+	else
+		echo "false"
+	fi
+}
+
+
+if [[ ! -f /boot/mac_addresses ]] ; then
+  echo "no mac address configuration file found .. exiting"
+  exit 1
+fi
+
+primary_mac=$(cat /boot/mac_addresses | awk -F= '/PRIMARY_MAC/ {print $2}' | tr '[A-Z]' '[a-z]')
+secondary_mac=$(cat /boot/mac_addresses | awk -F= '/SECONDARY_MAC/ {print $2}' | tr '[A-Z]' '[a-z]')
+
+default_device=""
+secondary_device=""
+default_connection_name=""
+secondary_connection_name=""
+
+for dev in $(nmcli device status | awk '/ethernet/ {print $1}'); do
+  dev_mac=$(nmcli -g GENERAL.HWADDR dev show $dev | sed -e 's/\\//g' | tr '[A-Z]' '[a-z]')
+  case $dev_mac in
+    $primary_mac)
+      default_device=$dev
+      default_connection_name=$(nmcli -g GENERAL.CONNECTION dev show $dev)
+      ;;
+    $secondary_mac)
+      secondary_device=$dev
+      secondary_connection_name=$(nmcli -g GENERAL.CONNECTION dev show $dev)
+      ;;
+    *)
+      ;;
+   esac
+done
+
+echo -e "default dev: $default_device ($default_connection_name)\nsecondary dev: $secondary_device ($secondary_connection_name)"
+
+if [[ $(is_con_exists "$default_connection_name") == "false" ]]; then
+	nmcli con add type ethernet \
+	              con-name "$default_connection_name" \
+	              conn.interface "$default_device" \
+	              connection.autoconnect no \
+	              ipv4.method auto
+fi
+if [[ $(is_con_active "$default_connection_name") == "false" ]]; then
+	nmcli con up "$default_connection_name"
+fi
+
+if [[ $(is_con_exists "$secondary_connection_name") == "false" ]]; then
+	nmcli con add type ethernet \
+	              con-name "$secondary_connection_name" \
+	              conn.interface "$secondary_device" \
+	              connection.autoconnect no \
+	              ipv4.method auto
```

Comment:
What about using disabled instead?

Reply:
Why should we? 
I mean "auto" properties such as "dns" and "routes" specify information that is added on to the information returned from automatic configuration

Reply:
I don't understand what you are saying. I think this should be "disabled" because of the goal of this script is to set the primary interface up and use for management, and to set the secondary interface down so it is not in a way.

Reply:
I was explained later on the goal of the script (I thought it was related to the old one) my bad :(

Reply:
no worries, it's good that we're keeping each other on check :)